SCMI requirements

vstehle opened the pull request:
On AArch64, when SCMI is implemented, in a table:
- Require minimum protocol versions (as in SCMI 2.0)
- Recommend minimum protocol versions (as in SCMI 3.2)

vstehle commented:
As for the SCMI Base protocol version rationale:
- Recent TF-A, OP-TEE, U-Boot and EDK II support 2.0.
- Recent Linux is happy with any version, prefers 2.1.

xypron commented on source/chapter3-secureworld.rst:
```diff
@@ -56,6 +56,26 @@ version compatible with caller version 1.2. [#FFANote]_
+On AArch64 platforms, when privileged or secure firmware implements the System
+Control and Management Interface (SCMI), it must conform to the Arm System
+Control and Management Interface specification [SCMI]_ and it must implement a
+Base protocol version 2.x, where x >= 1 is recommended. [#SCMINote]_
```
So Base protocol version 3.2 referenced by the SCMI link is explicitly forbiden (sic!).

xypron replied:
How about:
* Base protocol version >= 2.0, preferably >=2.1.
This would allow 3.7 too.

xypron reviewed (changes requested):
I guess you meant base protocol version >= 2.0 preferably >= 2.1 and not base protocol < 3.0.

vstehle commented:
Action for this pull request after [the EBBR call of Apr 23](https://github.com/ARM-software/ebbr/wiki/EBBR-Notes-2025.04.23): Add a comment why it cannot be 3.x and must be 2.x.

xypron commented:
LGTM

Reviewed-by: @xypron

etienne-lms commented on source/chapter3-secureworld.rst:
```diff
@@ -56,6 +56,30 @@ version compatible with caller version 1.2. [#FFANote]_
+.. [#SCMINote] SCMI is a set of software interfaces defined by Arm, used in
+   the system management of power domains, performances, clocks, sensors, reset,
+   voltage domains, power and pins.
+   SCMI v2.1 introduced the `NEGOTIATE_PROTOCOL_VERSION` command, which eases
```
I would suggest a explicit sentence *"SCMI specification v3.2 introduced the ..."*

vstehle commented:
Thank you @xypron for reviewing.
After [our call of May 7](https://github.com/ARM-software/ebbr/wiki/EBBR-Notes-2025.05.07), it seems that we still need to refine a few wordings here and there (thus moving back to "draft").

etienne-lms commented on source/chapter3-secureworld.rst:
```diff
@@ -56,6 +56,30 @@ version compatible with caller version 1.2. [#FFANote]_
+   SCMI v2.1 introduced the `NEGOTIATE_PROTOCOL_VERSION` command, which eases
+   agents and platforms interoperability.
+   As per the SCMI specification § 4.1.3 Protocol discovery, all agents relying
+   on version 2.x should interoperate with platforms implementing version 2.y,
```
I would remove this sentence and only mention the one below: *"Future SCMI specification versions 3.x ..."*.

etienne-lms commented on source/chapter3-secureworld.rst:
```diff
@@ -56,6 +56,30 @@ version compatible with caller version 1.2. [#FFANote]_
+   versions 3.x, after negotiation.
+
+.. warning:: A future version of this specification will require a minimum SCMI
+   Base protocol version of 2.1.
```
Here also, I would rather mention the SCMI spec version 3.2 where all protocols at least support their `NEGOTIATE_PROTOCOL_VERSION` command.

vstehle replied:
This should be resolved now.

vstehle commented:
Changes since May 7:
- Updated SCMI reference to v4.0 alp0 and updated our referenced chapter accordingly.
- Addressed two comments from @etienne-lms.

This is not ready yet as we need to decide if we require Base Protocol version or specification version, and for that we need a better understanding of what is actually implemented.

vstehle commented:
This is a bit refined now:
- Broaden the requirements to all cases of transport to encompass the case of a remote processor behind a mailbox.
- Consider all standard protocols; not only the base one.
- Convert to a table for clarity

The requirements correspond to spec version 2.0 B. They are supported by TF-A and (almost) by OP-TEE. They should be aligned with the (upcoming) Arm SystemReady SRS 3.1 and should be sufficient for Linux, which is very permissive.

The recommendations correspond to spec version 3.2 E. They should benefit to recent Linux kernels, which default to those versions and can negotiate to some extend.

xypron reviewed (approved):
LGTM

vstehle commented on source/chapter3-secureworld.rst:
```diff
@@ -73,6 +73,64 @@ defined in [PFDI]_ § 5 Compliance Requirements, and it must support a version
+     - 2.0
+     - 3.1
+   * - System power management
+     -
```
During [the EBBR call of Sep 24](https://github.com/ARM-software/ebbr/wiki/EBBR-Notes-2025.09.24), the general consensus is that mentioning v1.0 everywhere is less error-prone; I will update the table.

vstehle commented on source/references.rst:
```diff
@@ -41,6 +41,10 @@ Bibliography
    <https://developer.arm.com/documentation/den0022/fb>`_
    October 2024, `Arm Limited <https://www.arm.com/>`_
 
+.. [SCMI] `Arm System Control and Management Interface issue F (v4.0 alp0).
```
During [the EBBR call of Sep 24](https://github.com/ARM-software/ebbr/wiki/EBBR-Notes-2025.09.24) we came out with the following solution:
- Reference v3.2 and remove the system telemetry protocol from the table
- Do not mention the issue letter in the references anymore
I will update the reference.

vstehle commented:
Rebased, and in the mean time added one missing comma to the SCMI reference.

xypron commented:
I think this is good to merge. But I have not checked all the version numbers.

vstehle commented:
Accepted during [the EBBR call of Nov 5](https://github.com/ARM-software/ebbr/wiki/EBBR-Notes-2025.11.05).
